mcs/class/corlib/Test/System.Threading/WaitHandleTest.cs

```diff
@@ -395,6 +395,30 @@ public void InterrupedWaitOne ()
 			}
 		}
 
+		[Test]
+		public void InterruptedWaitOneWithTimeout ()
+		{
+			using (ManualResetEvent mre = new ManualResetEvent (false))
+			using (ManualResetEvent ready = new ManualResetEvent (false)) {
+				var thread = new Thread (() => {
+					ready.Set ();
+					mre.WaitOne (1000);
+				});
+
+				thread.Start ();
+				ready.WaitOne ();
+
+				Thread.Sleep (10); // wait a bit so we enter mre.WaitOne
+
+				DateTime start = DateTime.Now;
+				while (DateTime.Now < start.AddMilliseconds (500)) {
+					GC.Collect (1);
+				}
+
+				Assert.IsTrue (thread.Join (1000), "#2");
```

Comment:
the mre thread should run for 1000 milliseconds, and the main thread should first do other stuff for 500 milliseconds, then wait 1000 milliseconds. So the flakiness would happen if the mre thread waits more than 1500 milliseconds, which is exactly what this test is about.